src/legacy/status_im/chat/models/mentions.cljs

```diff
@@ -118,11 +125,28 @@
   {:events [:mention/on-change-text-success]}
   [{:keys [db]} result]
   (log/debug "[mentions] on-change-text-success" {:result result})
-  (let [{:keys [state chat-id mentionable-users input-segments]} (transfer-mention-result result)]
-    {:db (-> db
-             (assoc-in [:chats/mention-suggestions chat-id] mentionable-users)
-             (assoc-in [:chats/mentions chat-id :mentions] state)
-             (assoc-in [:chat/inputs-with-mentions chat-id] input-segments))}))
+  (let [{:keys [state chat-id mentionable-users
+                input-segments call-id call-time]} (transfer-mention-result result)]
+    (when (= call-id @current-call-id)
+      (debounce/debounce-and-dispatch [:mention/re-render-text-input
```

Comment:
Same here, the event handler must return nil or a map of two optional effects `:db` and `:fx`. `debounce-and-dispatch` is a stateful call.